functions/Get-DbaDbRole.ps1

```diff
@@ -0,0 +1,137 @@
+	Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+.NOTES
+Tags: Roles, Databases
+Author: Klaas Vandenberghe ( @PowerDBAKlaas )
+
+Website: https://dbatools.io
+Copyright: (C) Chrissy LeMaire, clemaire@gmail.com
+License: GNU GPL v3 https://opensource.org/licenses/GPL-3.0
+
+.LINK
+ https://dbatools.io/Get-DbaDbRole
+
+.EXAMPLE
+Get-DbaDbRole -SqlInstance ServerA
+
+Returns a custom object displaying SQLServer, Database, Role for all DatabaseRoles on sql instance ServerA.
+
+.EXAMPLE
+Get-DbaDbRole -SqlInstance ServerA | Out-Gridview
+
+Returns a gridview displaying SQLServer, Database, Role for all DatabaseRoles on sql instance ServerA.
+
+.EXAMPLE
+Get-DbaDbRole -SqlInstance ServerB\sql16 -ExcludeDatabase DBADB,TestDB
+
+Returns a gridview displaying SQLServer, Database, Role for DatabaseRoles on sql instance ServerB\sql16, except those in databases DBADB and TestDB.
+
+.EXAMPLE
+'ServerB\sql16','ServerA' | Get-DbaDbRole
+
+Returns a gridview displaying SQLServer, Database, Role for DatabaseRoles on sql instances ServerA and ServerB\sql16.
+
+.EXAMPLE
+Get-DbaDbRole -SqlInstance ServerB\sql16 -Database AccountingDB
+
+Returns a gridview displaying SQLServer, Database, Role for DatabaseRoles in database AccountingDB on sql instance ServerB\sql16.
+
+#>
+	[CmdletBinding()]
+	Param (
+		[parameter(Mandatory, ValueFromPipeline)]
+		[Alias('SqlServer', 'ServerInstance')]
+		[DbaInstanceParameter[]]$SqlInstance,
+		[Alias("Credential")]
+		[PSCredential]$SqlCredential,
+		[object[]]$Database,
+		[object[]]$ExcludeDatabase,
+		[switch]$NoFixedRole,
+		[switch][Alias('Silent')]$EnableException
+	)
+	
+	process {
+		
+		foreach ($Instance in $sqlInstance) {
+			try {
+				Write-Message -Level Verbose -Message "Connecting to $instance"
+				$server = Connect-SqlInstance -SqlInstance $instance -SqlCredential $SqlCredential
+			}
+			catch {
+				Stop-Function -Message "Failure" -Category ConnectionError -ErrorRecord $_ -Target $instance -Continue
+			}
+						
+			$dbs = $server.Databases
+			
+			if ($Database) {
+				Write-Message -Level Verbose -Message "Databases to check: $Database"
+				$dbs = $dbs | Where-Object Name -In $Database
+			}
+			
+			if ($ExcludeDatabase) {
+				Write-Message -Level Verbose -Message "Databases excluded from check: $ExcludeDatabase"
+				$dbs = $dbs | Where-Object Name -NotIn $ExcludeDatabase
+			}
+			
+			foreach ($db in $dbs) {
+				Write-Message -Level Verbose -Message "Checking accessibility of $db on $instance"
+				
+				if ($db.IsAccessible -ne $true) {
+					Write-Message -Level Warning -Message "Database $db on $instance is not accessible"
+					continue
+				}
+				
+				$dbroles = $db.roles
+				Write-Message -Level Verbose -Message "Getting Database Roles for $db on $instance"
+				
+				if ($NoFixedRole) {
+					$dbroles = $dbroles | Where-Object { $_.isfixedrole -eq $false }
```

Comment:
Can write this as `$dbroles = $dbroles | Where-Object IsFixedRole -EQ $false` to match pattern we've had in other commands.

Reply:
fixed